src/main/scala/uk/gov/nationalarchives/consignmentexport/Main.scala

```diff
@@ -29,8 +30,9 @@ object Main extends CommandIOApp("tdr-consignment-export", "Exports tdr files in
       case FileExport(consignmentId, taskToken) =>
         val exportFailedErrorMessage = s"Export for consignment $consignmentId failed"
         val stepFunction: StepFunction  = StepFunction(StepFunctionUtils(sfnAsyncClient(stepFunctionPublishEndpoint)))
-
+        def runHeartbeat(): IO[Unit] = stepFunction.sendHeartbeat(taskToken) >> IO.sleep(1 minute) >> runHeartbeat()
```

Comment:
What's the purpose of the `IO.sleep` and setting it to 1 minute?

Reply:
Cats effect doesn't have a proper scheduler and I didn't want to start including other libraries to do one so this function sends the heartbeat call, then sleeps for a minute, then calls itself again after that. This is all run in a separate thread so `IO.sleep` doesn't block the main thread.
Basically a primitive scheduler.